GetStream/stream-chat-android
# CAS-880 Call `UploadAttachmentsWorker` in `ChatDomain::retryMessages`

https://stream-io.atlassian.net/browse/CAS-880

### 🎯 Goal
Allow sending attachments while being offline.

### 🛠 Implementation details

Upload of pending attachments is done when the connection is recovered. When all pending attachments are uploaded the message is sent. 

### 🧪 Testing

1. Turn off data connection
2. Send message with attachments to a channel while being offline
3. Optional: kill the app and open it again
4. Turn on data connection
5. Attachments upload should resume and message should be send

### ☑️ Checklist


### 🎉 GIF

![giphy (1)](https://user-images.githubusercontent.com/4527432/120331309-ef22b900-c2ed-11eb-9f70-5ddc477f6961.gif)
